### DISCUSSION TITLE
Low voltage alarm at charging, settings and control hierarchy in the venus os.

### DISCUSSION THREAD

**Discussion by larsenf** [Louisvdw/dbus-serialbattery#529] [Q&A]
So I have a few questions that stem from the issue i am having, while charging either when i stop or at higher voltages i will get a low battery alarm and my multiplus-ii will cut ac output.

I would like to know the hierarchy of control within my system to help me undestand which settings I should be looking at to fix this issue. My system currently is as follows, raspi venus 2.93, jdb bms (with this driver) on a used 18s lipo4 diy 17kw pack, multiplus-ii 48/3000 120v, 48v-12v buck for DC system and 5.5kw diesel generator. No grid or solar for now.

Does the driver have any control over the jdb bms to shut on and off the discharge or charge, or does it only do so through the multi, leaving only my jdb app settings to control my dc system? With DVCC on do the settings within the multiplus-ii do anything, I would like to have them set at a lower total charge voltage so if the venus device isnt present and its on the grid it wont overcharge and trip the bms. Conversly I dont want the multiplus to trigger absorption early if im using the venus device and going off individual cell voltage.

What settings would effect my low voltage alarm triggering at higher voltage, after a while it will function normally again after multiple alarms within a minute or two. when the venus device is off it does not do this so i dont think its the multi. Is it possibly something to do with being at 100% SOC which it reaches before im actually at 100% due to what i assume is not being able to detects small current draw over time.

**Comment by larsenf** [comment]
ok im going to copy paste this from another thread to answer part of my question:

DVCC - Distributed Voltage and Current Control

Enabling DVCC changes a GX device from a passive monitor into an active controller.
With an Managed BMS battery using the serial connection and driver, the GX receives a Charge Voltage Limit (CVL), Charge Current Limit (CCL), Discharge Current Limit (DCL) from that battery and relays that to the connected inverter/chargers and solar chargers. These then disable their internal charge algorithms and simply do what they're told by the battery. There is no need to set-up charge voltages or choose the charge algorithm type.

In short if you don't enable DVCC then each device will produce power from the settings you have set them up with, while when you enable DVCC it tells the GX to use the settings that the battery provide to manage the system.

Example: You have set you MPPT to charge the battery with up to 70A and to a voltage of 56V. If you connect this to a battery that can only handle 50A and a voltage of 55V it will not know about this and the MPPT will charge to the values it was set up with.
If you enable DVCC the battery can tell the system it will only accept up to 50A charge current and the MPPT will override it's internal value and honor the 50A. Same with the the voltage which it will aim for 55V.

What the driver does is to use the values from the battery to keep the system inside the parameters that the battery can handle so that the battery does not need to disconnect itself to protect the cells (which means your power cuts out)

With DVCC disabled you will still see the values that the battery publish to the GX, but it will not be used for anything. It's just a display of numbers.

**Comment by mr-manuel** [comment]
Could you try this? https://github.com/Louisvdw/dbus-serialbattery/issues/526
